xt/lib/Pherkin/Extension/ledgersmb_steps/common_steps.pl

```diff
@@ -156,6 +156,45 @@
     $vendor->save;
 };
 
+Given qr/an unpaid AP transaction with "(.*)" for \$(\d+)$/, sub {
+    my $vendor_name = $1;
+    my $amount = $2;
+    my $dbh = S->{ext_lsmb}->admin_dbh;
+
+    my $q = $dbh->prepare("
+        INSERT INTO ap (invnumber, transdate, amount, netamount, duedate, curr, approved, entity_credit_account)
+        SELECT 'I-001', '2018-01-01', ?, ?, '2018-01-01', 'USD', TRUE, entity_credit_account.id
+        FROM entity
+        JOIN entity_credit_account ON (
+            entity_credit_account.entity_id = entity.id
+        )
+        WHERE entity.name = ?
+        AND entity.entity_class = 1
+        LIMIT 1
+        RETURNING ap.id
+    ");
+    $q->execute($amount, $amount, $vendor_name);
+    my $ap_id = $q->fetchrow_hashref()->{id};
+
+    $q = $dbh->prepare("
+        INSERT INTO acc_trans (trans_id, chart_id, amount, transdate)
+        SELECT ?, ?, ?, '2018-01-01'
+    ");
+
+    $q->execute(
+        $ap_id,
+        28, # 5700--Office Supplies
+        $amount * -1,
+    );
+
+    $q->execute(
+        $ap_id,
+        10, # 2100--Accounts Payable account
+        $amount,
+    );
```

Comment:
In PostgreSQL, `INSERT INTO ... VALUES` has been extended to support:
```sql
INSERT INTO table (field1, field2)
VALUES ('r1-f1', 'r1-f2'),
      ('r2-f1', 'r2-f2')
```

That way you can achieve the same in a more compact invocation (although compactness isn't a goal by itself, I thought I'd mention it, in case you didn't know).